components/authentication/authentication.js

```diff
@@ -99,11 +133,36 @@ function postForgotPassword(req,res){
     })
 }
 
+const validationGuard = (req,res, next) =>{
+    const errors = validationResult(req)
+    if(!errors.isEmpty())
+        return res.status(400).json({err : errors.array()})
+    next()
+}
 
+const userProfileValidator = [
+    body('email').isEmail(),
+    body('firstName').exists().trim().escape(),
+    body('lastName').exists().trim().escape(),
+    body('password').exists(),
+    body('dateOfBirth').exists().custom(
+        date =>{
+            const dateObject = dayjs(date)
+            return dayjs(dateObject, "MM-DD-YYYY", true).isValid()
+        }
+    ),
+    validationGuard
+]
+
+const loginCredentialsValidator = [
+    body('email').exists().isEmail(),
+    body('password').exists(),
+    validationGuard
+]

```

Comment:
We could move these to a file called `validators` ?

Reply:
They're now in a validator.js file within the same /authentication directory